From 4a5981b113d3e32bb46a9128a032f69c8be58306 Mon Sep 17 00:00:00 2001
From: Basique <basiqueevangelist@yandex.ru>
Date: Mon, 30 Dec 2024 20:54:37 +0300
Subject: [PATCH] add support for Alpine Linux packages (.apk)

Alpine Linux packages normally get misclassified as Android
packages, so this commit adds a mime type to disambiguate them.
---
 data/freedesktop.org.xml.in                      |   6 ++++++
 tests/mime-detection/list                        |   2 ++
 .../shared-mime-info-doc-2.4-r1.apk              | Bin 0 -> 2453 bytes
 3 files changed, 8 insertions(+)
 create mode 100644 tests/mime-detection/shared-mime-info-doc-2.4-r1.apk

diff --git a/data/freedesktop.org.xml.in b/data/freedesktop.org.xml.in
index ab739edb..f1d97020 100644
--- a/data/freedesktop.org.xml.in
+++ b/data/freedesktop.org.xml.in
@@ -1614,6 +1614,12 @@ command to generate the output files.
     <sub-class-of type="application/java-archive"/>
     <glob pattern="*.apk"/>
   </mime-type>
+  <mime-type type="application/x-alpine-package-keeper-package">
+    <comment>Alpine Package Keeper package</comment>
+    <generic-icon name="package-x-generic"/>
+    <sub-class-of type="application/x-compressed-tar"/>
+    <glob pattern="*.apk" weight="40"/>
+  </mime-type>
   <mime-type type="application/vnd.symbian.install">
     <comment>SIS package</comment>
     <acronym>SIS</acronym>
